pkg/expression/expr_to_pb_test.go

```diff
@@ -1590,7 +1591,11 @@ func TestExprPushDownToTiKV(t *testing.T) {
 	require.Len(t, pushed, len(exprs))
 	require.Len(t, remained, 0)
 	exprs = exprs[:0]
-	castByteAsStringFunc, err := NewFunction(mock.NewContext(), ast.Cast, types.NewFieldType(mysql.TypeString), byteColumn)
+	// when conv a column with type BIT, a cast function will be used to cast bit to a binary string
+	castTp := types.NewFieldType(mysql.TypeString)
+	castTp.SetCharset(charset.CharsetBin)
+	castTp.SetCollate(charset.CollationBin)
+	castByteAsStringFunc, err := NewFunction(mock.NewContext(), ast.Cast, castTp, byteColumn)
```

Comment:
This function can not be push down to TiKV because `from_binary` can not be push down to TiKV?

Reply:
no, I think it's the issue: https://github.com/pingcap/tidb/issues/51877 The tidb have some special logic for function `conv`. When converting a BIT type, it uses `conv(cast(bit as char))` to cast bit first, but `conv` should also handles the result string as a number in binary format instead of string... 

See: https://github.com/pingcap/tidb/blob/9530fdc2388d025af86eb6fce5e617972c361000/pkg/expression/builtin_math.go#L1281-L1290

Reply:
I added a comment in this test